suite/suite.go

```diff
@@ -52,12 +77,19 @@ func (suite *Suite) Require() *require.Assertions {
 // can call `suite.Assert().NoError()`.
 func (suite *Suite) Assert() *assert.Assertions {
 	if suite.Assertions == nil {
-		suite.Assertions = assert.New(suite.T())
+		var tT assert.TestingT
+		if suite.T() != nil {
+			tT = suite.T()
+		}
+		if suite.F() != nil {
+			tT = suite.F()
+		}
+		suite.Assertions = assert.New(tT)
 	}
 	return suite.Assertions
 }
 
-func failOnPanic(t *testing.T) {
+func failOnPanic(t testing.TB) {
```

Comment:
What's the logic for expanding the interface to include benchmarks and removing the pointer here?

Reply:
The reason for replacing `*testing.T` by `testing.TB` is so that `failOnPanic()` can be called with a `testing.F` value as well (`testing.TB` interface is satisfied by `T`, `F` and `B` [source](https://sourcegraph.com/github.com/golang/go/-/blob/src/testing/testing.go?L743))